great_expectations/datasource/data_connector/util.py

```diff
@@ -342,29 +345,41 @@ def sanitize_prefix_for_s3(text: str) -> str:
 
 
 def normalize_directory_path(
-    dir_path: str, root_directory_path: Optional[str] = None
-) -> str:
+    dir_path: Union[pathlib.Path, str],
```

Comment:
This utility method, developed as part of V3, was updated to serve both V3 and ZEP (where paths are handled more formally).